plugins/woocommerce/src/StoreApi/docs/checkout.md

```diff
@@ -65,6 +65,84 @@ curl --header "Nonce: 12345" --request GET https://example-store.com/wp-json/wc/
 }
 ```
 
+## Update checkout data
+
+This endpoint allows you to update the checkout data for the current order. This can be called from the frontend to persist checkout fields, for example.
+
+```http
+PUT /wc/store/v1/checkout?__experimental_calc_totals=true
+```
+
+Note the `__experimental_calc_totals` parameter. This is used to determine if the cart totals should be recalculated. This should be set to true if the cart totals are being updated in response to a PUT request, false otherwise.
+
+| Attribute           | Type   | Required | Description                                         |
+| :------------------ | :----- | :------: | :-------------------------------------------------- |
+| `additional_fields` | object |    No    | Name => value pairs of additional fields to update. |
+| `payment_method`    | string |    No    | The ID of the payment method selected.              |
```

Comment:
It looks like [this endpoint accepts `order_notes`](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/src/StoreApi/Routes/V1/Checkout.php#L128) too, which is omitted from this table.

Reply:
Added in https://github.com/woocommerce/woocommerce/pull/56906/commits/47f860f864eeaa64d1957e000ab50d68946a1b96